simple_parsing/parsing.py

```diff
@@ -159,6 +166,11 @@ def __init__(
         if add_config_path_arg is None:
             # By default, add a config path argument if a config path was passed.
             add_config_path_arg = bool(config_path)
+        if isinstance(add_config_path_arg, str) and not add_config_path_arg.isidentifier():
+            raise ValueError(
+                "If `add_config_path_arg` is a string it must be a valid Python identifier (no dashes)."
+                f" Not: {add_config_path_arg}"
+            )
```

Comment:
Actually, you can use attribute names with dashes when you `getattr` and `setattr` on a `Namespace` object:
```python
>>> from argparse import Namespace
>>> ns = Namespace()
>>> setattr(ns, "some-attr", 42)
>>> getattr(ns, "some-attr")
42
```

Reply:
> Actually, you can use attribute names with dashes when you `getattr` and `setattr` on a `Namespace` object:
> 
> ```python
> >>> from argparse import Namespace
> >>> ns = Namespace()
> >>> setattr(ns, "some-attr", 42)
> >>> getattr(ns, "some-attr")
> 42
> ```

Thanks good idea. I have it working now. For reference, you have to replace the dashes with underscores like so:
```python
getattr(args_with_config_path, config_path_name.replace('-', '_'))
```

Reply:
> Also, if you have the time, would you mind adding a simple test so we can cover this change?
> 
> ```python
> @pytest.mark.parametrize("invalid_value", ["foo.bar.baz?", "-------", "bob bob bob"])  # add other more relevant examples perhaps?
> def test_pass_invalid_value_to_add_config_path_arg(invalid_value):
>     with pytest.raises(ValueError):
>         _ = ArgumentParser(add_config_path_arg=invalid_value)
> ```

I've added some tests